src/sentry/search/eap/columns.py

```diff
@@ -261,31 +274,32 @@ class ConditionalAggregateDefinition(FunctionDefinition):
 
     # The type of aggregation (ex. sum, avg)
     internal_function: Function.ValueType
-    # The attribute to conditionally aggregate on
-    key: AttributeKey
-    # A function that takes in the resolved argument and returns the condition to filter on
-    filter_resolver: Callable[..., TraceItemFilter]
+    # A function that takes in the resolved argument and returns the condition to filter on and the key to aggregate on
+    aggregate_resolver: Callable[..., tuple[AttributeKey, TraceItemFilter]]
 
     def resolve(
-        self, alias: str, search_type: constants.SearchType, resolved_argument: AttributeKey | None
+        self,
+        alias: str,
+        search_type: constants.SearchType,
+        resolved_arguments: list[AttributeKey | Any],
     ) -> ResolvedConditionalAggregate:
+        (key, filter) = self.aggregate_resolver(*resolved_arguments)
```

Comment:
```suggestion
        key, filter = self.aggregate_resolver(*resolved_arguments)
```

Reply:
🤔 I'm worried that `*resolved_arguments` is going to end up being a footgun, can we change `aggregate_resolver` to `Callable[List[AttributeKey | Any], tuple[AttributeKey, TraceItemFilter]]` and just pass `resolved_arguments` as-is instead?